app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileHelper.java

```diff
@@ -0,0 +1,129 @@
+package com.nervousfish.nervousfish.activities;
+
+import android.graphics.Color;
+import android.widget.EditText;
+
+import com.nervousfish.nervousfish.data_objects.IKey;
+import com.nervousfish.nervousfish.data_objects.KeyPair;
+import com.nervousfish.nervousfish.data_objects.SimpleKey;
+import com.nervousfish.nervousfish.modules.cryptography.IKeyGenerator;
+
+/**
+ * Helper method for the logical functionality of the {@link CreateProfileActivity}.
+ */
+class CreateProfileHelper {
+
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final String DEFAULT_KEY_NAME = "NervousFish generated key";
+
+    private final IKeyGenerator keyGenerator;
+    private final int alertColor;
+
+    /**
+     * Helper class for the create profile {@link android.app.Activity}
+     * containing its logical functionality.
+     *
+     * @param keyGenerator A {@link IKeyGenerator} to generate keys.
+     * @param alertColor The {@link Color} to set as alert for incorrect values.
+     */
+    CreateProfileHelper(final IKeyGenerator keyGenerator, final int alertColor) {
+        this.keyGenerator = keyGenerator;
+        this.alertColor = alertColor;
+    }
+
+    /**
+     * Generates a KeyPair based on the type selected.
+     *
+     * @param keyType The type of key to generate.
+     * @return a {@link KeyPair} with the key type selected
+     */
+    KeyPair generateKeyPair(final IKey.Types keyType) {
+        switch (keyType) {
+            case RSA:
+                return this.keyGenerator.generateRSAKeyPair(CreateProfileHelper.DEFAULT_KEY_NAME);
+            case Simple:
+                final IKey publicKey = new SimpleKey("public", "foo");
```

Comment:
> Wasn't simple key meant as a single string to both encrypt and decrypt a message instead of using a private and public key?

No it wasn't, it was meant as an example implementation of `IKey`


> I cannot imagine any reason why anybody would ever use it, so maybe it's better to replace simple key with DSA which would make a lot more sense imo.

True, but we don't have an implementation of DSA yet so I couldn't add a case for that.


I'll create a new issue for this so that we don't forget about it 😄